plasmapy/plasma/grids.py

```diff
@@ -336,6 +419,29 @@ def add_quantity(self, key: str, quantity: u.Quantity):
         data = xr.DataArray(quantity, dims=axes, attrs={"unit": quantity.unit})
         self.ds[key] = data
 
+    def add_quantities(self, keys: list, quantities: list):
+        r"""
+        Adds a list of keys and quantities to the grid. See "add_quantity"
+
+        Parameters
+        ----------
+        keys : list
+            A list of str keys for quantities to add
+        quantities : list
+            A list of quantities to add. Each must be an array that matches
+            the size of the grid
```

Comment:
This smells like it should take a single dictionary argument instead, doesn't it? Maybe even something like
```
    def add_quantities(self, **quantities: dict):
        r"""
        Adds a list of keys and quantities to the grid. See "add_quantity"

        Parameters
        ----------
        quantities :
            quantities to add. Each must be an array that matches
            the size of the grid. The argument names should fit `recognized_keys`
```

But I'm not sure how much I like that versus the single dictionary option, so I'm not making this a "suggestion" yet. I'll note that a single dictionary could then be passed in as `grid.add_quantities(**my_awesome_dictionary)`.

... come to think of it, I think this option would make `add_quantity` redundant; `grid.add_quantity("E_x", E_x_array)` would be equivalent to `grid.add_quantities(E_x=E_x_array)`. The naming might then use adjustment, of course.

Reply:
You're absolutely right - consider it done.  I'm going with the option `grid.add_quantities(E_x=E_x_array)` for now.